packages/app/src/systems/Core/hooks/useFuel.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from 'react';
+
+const globalWindow = typeof window !== 'undefined' ? window : ({} as Window);
+
+export function useFuel() {
+  const [error, setError] = useState('');
+  const [isLoading, setLoading] = useState(true);
+  const [fuel, setFuel] = useState<Window['fuel']>(globalWindow.fuel);
+
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      if (globalWindow.fuel) {
+        setFuel(globalWindow.fuel);
+      } else {
+        setError('fuel not detected on the window!');
+      }
+      setLoading(false);
+    }, 500);
+    return () => clearTimeout(timeout);
+  }, []);
```

Comment:
Take a look on the docs to implement using the latest version of this, wich includes using FuelLoaded event.